src/data_automate/data_availability.py

```diff
@@ -0,0 +1,110 @@
+import gcsfs
+import logging
+
+import typing as t
+
+from arco_era5 import (
+    SINGLE_LEVEL_VARIABLES,
+    MULTILEVEL_VARIABLES,
+    PRESSURE_LEVELS_GROUPS,
+)
+
+logger = logging.getLogger(__name__)
+
+# File Templates
+MODELLEVEL_DIR_VAR_TEMPLATE = (
+    "gs://gcp-public-data-arco-era5/raw/ERA5GRIB/HRES/Daily/{year:04d}/{year:04d}{month:02d}{day:02d}_hres_{chunk}.grb2_{level}_{var}.grib")
+MODELLEVEL_DIR_TEMPLATE = (
+    "gs://gcp-public-data-arco-era5/raw/ERA5GRIB/HRES/Daily/{year:04d}/{year:04d}{month:02d}{day:02d}_hres_{chunk}.grb2")
+SINGLELEVEL_DIR_VAR_TEMPLATE = (
+    "gs://gcp-public-data-arco-era5/raw/ERA5GRIB/HRES/Month/{year:04d}/{year:04d}{month:02d}_hres_{chunk}.grb2_{level}_{var}.grib")
+SINGLELEVEL_DIR_TEMPLATE = (
+    "gs://gcp-public-data-arco-era5/raw/ERA5GRIB/HRES/Month/{year:04d}/{year:04d}{month:02d}_hres_{chunk}.grb2")
+PRESSURELEVEL_DIR_TEMPLATE = (
+    "gs://gcp-public-data-arco-era5/raw/date-variable-pressure_level/{year:04d}/{month:02d}/{day:02d}/{chunk}/{pressure}.nc")
+AR_SINGLELEVEL_DIR_TEMPLATE = (
+    "gs://gcp-public-data-arco-era5/raw/date-variable-single_level/{year:04d}/{month:02d}/{day:02d}/{chunk}/surface.nc")
+
+# Data Chunks
+model_level_chunks = ["dve", "tw", "o3q", "qrqs"]
+single_level_chunks = [
+    "cape", "cisst", "sfc", "tcol", "soil_depthBelowLandLayer_istl1",
+    "soil_depthBelowLandLayer_istl2", "soil_depthBelowLandLayer_istl3",
+    "soil_depthBelowLandLayer_istl4", "soil_depthBelowLandLayer_stl1",
+    "soil_depthBelowLandLayer_stl2", "soil_depthBelowLandLayer_stl3",
+    "soil_depthBelowLandLayer_stl4", "soil_depthBelowLandLayer_swvl1",
+    "soil_depthBelowLandLayer_swvl2", "soil_depthBelowLandLayer_swvl3",
+    "soil_depthBelowLandLayer_swvl4", "soil_surface_tsn", "lnsp",
+    "zs", "rad", "pcp_surface_cp", "pcp_surface_crr",
+    "pcp_surface_csf", "pcp_surface_csfr", "pcp_surface_es",
+    "pcp_surface_lsf", "pcp_surface_lsp", "pcp_surface_lspf",
+    "pcp_surface_lsrr", "pcp_surface_lssfr", "pcp_surface_ptype",
+    "pcp_surface_rsn", "pcp_surface_sd", "pcp_surface_sf",
+    "pcp_surface_smlt", "pcp_surface_tp"]
+pressure_level_chunks = MULTILEVEL_VARIABLES
+ar_single_level_chunks = SINGLE_LEVEL_VARIABLES
+pressure_level = PRESSURE_LEVELS_GROUPS["full_37"]
+
+
+def check_data_availability(co_date_range: t.List, ar_date_range: t.List):
```

Comment:
A list of what?

Reply:
Can we make it return a bool (instead of 1 and 0)? Please add this annotation, too.

Reply:
A list of datetime objects.
All other changes were done. Thank you for the suggestions.